### PR TITLE
Stack LSTM Net for Paddle Book6

### LINKED ISSUE #5504: stack_lstm_net for book6

### PR DESCRIPTION
**Pull request by QiJune**
Fix #5504

### CODE REVIEW

File: python/paddle/v2/framework/layers.py [hunk 1]
```diff
@@ -81,6 +81,55 @@ def embedding(input,
...
+            'Cell': cell,
+            'BatchGate': batch_gate,
+            'BatchCellPreAct': batch_cell_pre_act
+        },
+        attrs={
+            'usePeepholes': use_peepholes,
+            'isReverse': is_reverse,
+            'gateActivation': gate_activation,
+            'cellActivation': cell_activation,
+            'candidateActivation': candidate_activation
```
Comment (qingqing01) [inline python/paddle/v2/framework/layers.py]: these attr names have been changed to `snake_case`, please update.

File: python/paddle/v2/framework/layers.py [hunk 2]
```diff
@@ -81,6 +81,55 @@ def embedding(input,
...
+        })
+    return hidden
```
Comment (qingqing01) [inline python/paddle/v2/framework/layers.py]: The `Cell` is also the output.

File: python/paddle/v2/framework/tests/test_understand_sentiment_dynamic_lstm.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    # add bias attr
+
+    # TODO(qijun) linear act
+    fc1 = layers.fc(input=emb, size=hid_dim)
+    lstm1 = layers.dynamic_lstm(input=fc1, size=hid_dim)
+
+    inputs = [fc1, lstm1]
+
+    for i in range(2, stacked_num + 1):
+        fc = layers.fc(input=inputs, size=hid_dim)
```
Comment (qingqing01) [inline python/paddle/v2/framework/tests/test_understand_sentiment_dynamic_lstm.py]: 这处和book不一致，book:  https://github.com/PaddlePaddle/book/blob/develop/06.understand_sentiment/train.py#L58  

这个fc有两个输入，有两组weight，每个weight的初始化，强调下lstm作为输入的weight初始化是0。

```python
fc_para_attr = paddle.attr.Param(learning_rate=1e-3)
lstm_para_attr = paddle.attr.Param(initial_std=0., learning_rate=1.)
```

File: python/paddle/v2/framework/tests/test_understand_sentiment_dynamic_lstm.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        lstm = layers.dynamic_lstm(
+            input=fc, size=hid_dim, is_reverse=(i % 2) == 0)
```
Comment (qingqing01) [inline python/paddle/v2/framework/tests/test_understand_sentiment_dynamic_lstm.py]: 这里有一处和book不同：lstm的`candidate_activation`在book里(即book里的act)用的是`relu`，

https://github.com/PaddlePaddle/book/blob/develop/06.understand_sentiment/train.py#L80

File: python/paddle/v2/framework/tests/test_understand_sentiment_dynamic_lstm.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        inputs = [fc, lstm]
+
+    fc_last = layers.sequence_pool(input=inputs[0], pool_type='max')
+    lstm_last = layers.sequence_pool(input=inputs[1], pool_type='max')
+
+    prediction = layers.fc(input=[fc_last, lstm_last],
+                           size=class_dim,
+                           act='softmax')
+    cost = layers.cross_entropy(input=prediction, label=label)
```
Comment (qingqing01) [inline python/paddle/v2/framework/tests/test_understand_sentiment_dynamic_lstm.py]: 为了数值稳定性，我们有[`softmax_with_cross_entropy_op`]( https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/softmax_with_cross_entropy_op.h)，建议demo里 `softmax+ cross_entropy`换成`softmax_with_cross_entropy_op` ?

File: python/paddle/v2/framework/tests/test_understand_sentiment_dynamic_lstm.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    dict_dim = len(word_dict)
+    class_dim = 2
+
+    cost, acc = stacked_lstm_net(input_dim=dict_dim, class_dim=class_dim)
+
+    train_data = paddle.batch(
+        paddle.reader.shuffle(
+            paddle.dataset.imdb.train(word_dict), buf_size=1000),
+        batch_size=BATCH_SIZE)
+    place = core.CPUPlace()
```
Comment (qingqing01) [inline python/paddle/v2/framework/tests/test_understand_sentiment_dynamic_lstm.py]: 作为test是不是gpu更快些？

File: python/paddle/v2/framework/tests/test_understand_sentiment_dynamic_lstm.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+            label = np.array(map(lambda x: x[1], data)).astype("int64")
+            label = label.reshape([BATCH_SIZE, 1])
+
+            tensor_label = core.LoDTensor()
+            tensor_label.set(label, place)
+
+            outs = exe.run(g_main_program,
+                           feed={"words": tensor_words,
+                                 "label": tensor_label},
+                           fetch_list=[cost, acc])
```
Comment (qingqing01) [inline python/paddle/v2/framework/tests/test_understand_sentiment_dynamic_lstm.py]: 后续会作为demo吗？ 如果作为demo，是不是应该测试下test集？(也可以加TODO，作为后续PR。)
